sonic-chassisd/scripts/chassisshow

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+"""
+    chassisshow
+    Chassis information update tool for SONiC
+    This tool runs one time at the launch of the platform monitor in order to populate STATE_DB with chassis information such as model, serial number, and revision.
+"""
+
+try:
+    import os
+    import sys
+
+    from sonic_py_common import daemon_base, logger
+
+    # If unit testing is occurring, mock swsscommon and module_base
+    if os.getenv("CHASSISSHOW_UNIT_TESTING") == "1":
+        from tests import mock_swsscommon as swsscommon
+    else:
+        from swsscommon import swsscommon
+except ImportError as e:
+    raise ImportError(str(e) + " - required module not found")
+
+#
+# Constants
+#
+
+SYSLOG_IDENTIFIER = "chassisshow"
+
+CHASSIS_INFO_TABLE = 'CHASSIS_INFO'
+CHASSIS_INFO_KEY_TEMPLATE = 'chassis {}'
+CHASSIS_INFO_CARD_NUM_FIELD = 'module_num'
+CHASSIS_INFO_SERIAL_FIELD = 'serial'
+CHASSIS_INFO_MODEL_FIELD = 'model'
+CHASSIS_INFO_REV_FIELD = 'revision'
+
+CHASSIS_LOAD_ERROR = 1
+
+#
+# Functions
+# 
+
+def provision_db(platform_chassis, log):
+    # Init state db connection
+    state_db = daemon_base.db_connect("STATE_DB")
+    chassis_table = swsscommon.Table(state_db, CHASSIS_INFO_TABLE)
+
+    # Populate DB with chassis hardware info 
+    fvs = swsscommon.FieldValuePairs([
+                                        (CHASSIS_INFO_SERIAL_FIELD, platform_chassis.get_serial()),
```

Comment:
In the case these APIs are not supported (by other vendors) what will be populated?

Reply:
I need to pull in the `try_get` function from `chassisd` to wrap this. Good catch thank you. Right now it throws an unhandled NotImplementedError.